src/core/namespace.c

```diff
@@ -1634,7 +1634,7 @@ static int apply_one_mount(
 
                         q = make_mount_point_inode_from_path(what, mount_entry_path(m), 0755);
                         if (q < 0 && q != -EEXIST)
-                                log_error_errno(q, "Failed to create destination mount point node '%s': %m",
+                                log_debug_errno(q, "Failed to create destination mount point node '%s': %m",
```

Comment:
I had left this intentionally at error level, because it is usually a user configuration error and it is a gigantic pain to debug looking at past logs (can't retroactively change log level), so please leave it and maybe add a comment explaining the utility

Reply:
that's a hack though. please fix properly, i.e. return recognizable error, and propagate file path via setup_namespace()'s `error_path` parameter.

Reply:
That doesn't work, there's no recognizable error that can be returned that says it was trying to create a mount point, errnos are too generic

Reply:
we return recognizable errors all the time, just do `git grep recognizable` over the tree. i.e. return `-ENOANO` or something like this, and then make the caller look for that.

yeah, errnos are shit, but it's how things are done in our codebase.

Reply:
There is no law of physics that says you can't log - it's a cosmetic stylistic choice, which is valid and reasonable in 99.999% of cases. But sometimes it is not, and causes real problems. If you know of a way to make these errors recognizable and useful then please do so, but just making things more difficult for real-world users for a cosmetic change is not a good trade-off I'm sure. Having things like "foo.service: Failed at step NAMESPACE spawning /bin/bash: No such file or directory" when /bin/bash is obviously there just makes for an unreasonably bad user experience

Reply:
hmm, that's not really when this message is printed. if the mount inode already exists we'll see EEXIST here, which is excluded from this log message.

Reply:
Sure, it was just a random example to illustrate the point. With the current changes, the combination of "Failed at step NAMESPACE spawning /bin/bash:" plus whatever errno comes back from these systems call is not recognizable and is interpreted as a systemd bug. If you know how to make it individually recognizable from that single log_error call, then please do so

Reply:
> Having things like "foo.service: Failed at step NAMESPACE spawning /bin/bash: No such file or directory" when /bin/bash is obviously there just makes for an unreasonably bad user experience

the log messages don't look like that anymore, since 7cc5ef5f1811c539ae7f20255c2a093f413cc64f...

There's now a message: `Failed to set up mount namespacing: /some/path: No such file or directory.`

Reply:
Yes, which is still not useful. Was it creating a mount point? Was it looking for the source? Etc etc

Reply:
you know if you want more details, then *debug* logs exist! That' what it is for. you get a rough idea without them, but if you want all possibly details, then turn up the log level. It think that concept is actually widely understood in the industry.

Reply:
Also, why even LOG_ERR? right now the error is just ignored...

Reply:
also, outputing the path as it is here is not cool, according to a certain @bluca, because that's what 13b3af4aa2f83b8ff853a63fe21001d170a5cfb5 corrected...

Reply:
> you know if you want more details, then _debug_ logs exist! That' what it is for. you get a rough idea without them, but if you want all possibly details, then turn up the log level. It think that concept is actually widely understood in the industry.

It's not all possible details, it's highlighting the ones coming from configuration mistakes, according to what I've seen happen in the real world. You cannot have debug level always enabled or it would flood the tables and literally cost millions to store.

> also, outputing the path as it is here is not cool, according to a certain @bluca, because that's what [13b3af4](https://github.com/systemd/systemd/commit/13b3af4aa2f83b8ff853a63fe21001d170a5cfb5) corrected...

That's about hiding the leading working directory, which is just confusing, so for example if you have BindPaths=/var/foo the error string says '/var/foo' instead of '/run/systemd/unit-root/var/foo' which is an internal detail, and _that_ doesn't make sense to have in a general log message.